Content.Server/Sports/PitchingMachineSystem.cs

```diff
@@ -0,0 +1,152 @@
+
+                if (ballLauncher.IsOn)
+                    Fire(ballLauncher.Owner);
+            }
+        }
+
+        private void OnInteractUsing(EntityUid uid, PitchingMachineComponent component, InteractUsingEvent args)
+        {
+            if (!TryComp<BallisticAmmoProviderComponent>(component.Owner, out var ammoProviderComponent))
+                return;
+
+            if (HasComp<ToolComponent>(args.Used))
+                return;
+
+            if (ammoProviderComponent.Capacity == ammoProviderComponent.Entities.Count + 1)
+                return;
+
+            ammoProviderComponent.Entities.Add(args.Used);
+            ammoProviderComponent.Container.Insert(args.Used);
+        }
+
+        private void OnPowerChanged(EntityUid uid, PitchingMachineComponent component, PowerChangedEvent args)
+        {
+            if (!args.Powered)
+            {
+                component.IsOn = false;
+            }
+        }
+
+        public void TogglePower(EntityUid uid, PitchingMachineComponent component)
+        {
+            if (!TryComp<ApcPowerReceiverComponent>(component.Owner, out var powerReceiverComponent) || !powerReceiverComponent.Powered)
+                return;
+
+            if (!component.IsOn)
+            {
+                component.IsOn = true;
+                _popupSystem.PopupEntity(Loc.GetString("comp-emitter-turned-on", ("target", component.Owner)), component.Owner, Filter.Pvs(component.Owner));
+            }
+            else
+            {
+                component.IsOn = false;
+                _popupSystem.PopupEntity(Loc.GetString("comp-emitter-turned-off", ("target", component.Owner)), component.Owner, Filter.Pvs(component.Owner));
+            }
+        }
+
+        private void Fire(EntityUid uid)
+        {
+            if (!TryComp<PitchingMachineComponent>(uid, out var component))
+                return;
+            if (!TryComp<BallisticAmmoProviderComponent>(component.Owner, out var ammoProviderComponent))
+                return;
+            if (ammoProviderComponent.Container.ContainedEntities.Count == 0 || ammoProviderComponent.Entities.Count == 0)
+                return;
+
+            var projectile = _robustRandom.Pick(ammoProviderComponent.Container.ContainedEntities);
+
+            var dir = Comp<TransformComponent>(component.Owner).WorldRotation.ToWorldVec() * _robustRandom.NextFloat(component.ShootDistanceMin, component.ShootDistanceMax);
+
+            //manually shooting it because guns can't seem to shoot themselves
+            ammoProviderComponent.Entities.Remove(projectile);
+            ammoProviderComponent.Container.Remove(projectile);
+
+            _audioSystem.Play(_audioSystem.GetSound(component.FireSound), Filter.Pvs(component.Owner), component.Owner, AudioParams.Default);
+
+            _throwingSystem.TryThrow(projectile, dir, 10f); //using throw because guncode shootprojectile is hardcoded to absolutely yeet it, impossible to actually hit with a bat
```

Comment:
`_gunSystem.ShootProjectile()` now supports setting a custom velocity, so you can use that for this.